pygeoapi/provider/mvt_postgres.py

```diff
@@ -0,0 +1,263 @@
+
+    @property
+    def service_url(self):
+        return self._service_url
+
+    @property
+    def service_metadata_url(self):
+        return self._service_metadata_url
+
+    def get_layer(self):
+        """
+        Extracts layer name from url
+
+        :returns: layer name
+        """
+
+        return self.layer_name
+
+    def get_tiling_schemes(self):
+
+        return [
+                TileMatrixSetEnum.WEBMERCATORQUAD.value,
+                TileMatrixSetEnum.WORLDCRS84QUAD.value
+            ]
+
+    def get_tiles_service(self, baseurl=None, servicepath=None,
+                          dirpath=None, tile_type=None):
+        """
+        Gets mvt service description
+
+        :param baseurl: base URL of endpoint
+        :param servicepath: base path of URL
+        :param dirpath: directory basepath (equivalent of URL)
+        :param tile_type: tile format type
+
+        :returns: `dict` of item tile service
+        """
+
+        super().get_tiles_service(baseurl, servicepath,
+                                  dirpath, tile_type)
+
+        self._service_url = servicepath
+        return self.get_tms_links()
+
+    def get_tiles(self, layer=None, tileset=None,
+                  z=None, y=None, x=None, format_=None):
+        """
+        Gets tile
+
+        :param layer: mvt tile layer
+        :param tileset: mvt tileset
+        :param z: z index
+        :param y: y index
+        :param x: x index
+        :param format_: tile format
+
+        :returns: an encoded mvt tile
+        """
+        if format_ == 'mvt':
+            format_ = self.format_type
+
+        fields_arr = self.postgres.get_fields().keys()
+        fields = ', '.join(['"' + f + '"' for f in fields_arr])
+        if len(fields) != 0:
+            fields = ',' + fields
+        
+        query = ''
+        if tileset == TileMatrixSetEnum.WEBMERCATORQUAD.value.tileMatrixSet:
+            query = text("""
+                WITH                                                                  
+                    bounds AS (
+                        SELECT ST_TileEnvelope(:z, :y, :x) AS boundgeom
+                    ),
+                    mvtgeom AS (
+                        SELECT ST_AsMVTGeom(ST_Transform({geom}, 3857), bounds.boundgeom) AS geom {fields}
+                        FROM "{table}", bounds
+                        WHERE ST_Intersects({geom}, ST_Transform(bounds.boundgeom, 4326))
+                    )
+                SELECT ST_AsMVT(mvtgeom, 'default') FROM mvtgeom;
+            """.format(geom=self.geom, table=self.table, fields=fields))  
+        
+        if tileset == TileMatrixSetEnum.WORLDCRS84QUAD.value.tileMatrixSet:
+            query = text("""
+                WITH                                                                  
+                    bounds AS (
+                        SELECT ST_TileEnvelope(:z, :y, :x, ST_MakeEnvelope(-180, -90, 180, 90, 4326)) AS boundgeom
```

Comment:
Why the tile envelope here is based on the extent of the crs, instead of the extent of the geometry (like in the case of WebMercator)?

Reply:
[ST_TileEnvelope](https://postgis.net/docs/ST_TileEnvelope.html) by default is configured to EPSG:3857. In order to use it for other CRSs, the docs mention that you can add an extra parameter of a geometry that has an extent of the 0th zoom level. There's an example of setting it for EPSG:4326, we just used that here. (hope that this answered your question, if not, please let me know)

We can probably use the WKB representation there so that `MakeEnvelope` isn't calculated for every tile request.

Reply:
I agree, I think the WKB representation would make sense here!

Reply:
Addressed in [97ba024](https://github.com/geopython/pygeoapi/pull/1979/commits/97ba024f4aa8a253ff04ac718bfb3feba528080a). I used EWKT representation instead of WKB as the the WKB hex string was too long